### PR TITLE
[actions] Add notification builders & Mark some actions deprecated as replaced by better APIs

### PR DESCRIPTION
**Pull request by florian-h05**
This adds builders to create broadcast, log and standard notifications and allow easy usage of the new functionality provided by https://github.com/openhab/openhab-addons/issues/16934 without having to deal with method param orders.

It also marks several actions as deprecated, because their functionality was replaced by better and often pure JS APIs.

### CODE REVIEW

**Comment by florian-h05** [comment]
To Dos:

- [ ] Add JSDoc for onClickAction, actionButton1 etc. once action format docs have been released.
- [ ] Extend README for onClickAction, actionButton1 etc. once action format docs have been released.
- [ ] Regenerate type defs.

**Comment by digitaldan** [comment]
Awesome !

File: README.md [hunk 1]
```diff
@@ -792,21 +782,50 @@ See [openhab-js : actions.Voice](https://openhab.github.io/openhab-js/actions.ht
...
+- Broadcast Notifications: Sent to all registered devices and shown as notification on these devices.
+- Log Notifications: Only shown in the notification log, e.g. inside the Android and iOS Apps.
+- Standard Notifications: Sent to the registered devices of the specified user and shown as notification on his devices.
+
+To send these three types of notifications, two notification builders are available:
+
+- `.logNotificationBuilder(message)`: Creates a builder to send a log notification with the given message.
+  - `.withIcon(icon)`: Sets the icon of the notification.
+  - `.withSeverity(link)`: Sets the severity of the notification.
+  - .`send()`: Sends the notification.
```
Comment (mueller-ma) [inline README.md:800]: I'd replace this builder with a function `logOnly()` on `notificationBuilder()`. This way the media attachment can be shown in the notification list.

File: README.md [hunk 2]
```diff
@@ -792,21 +782,50 @@ See [openhab-js : actions.Voice](https://openhab.github.io/openhab-js/actions.ht
...
+- `.notificationBuilder(message)`: Creates a builder to send a broadcast or standard notification with the given message.
+  - all methods of the log notification builder plus:
+  - `.withUserId(userId)`: By specifying the mail address of a specific openHAB Cloud user, the notification is sent only to this user.
```
Comment (mueller-ma) [inline README.md:803]: What about `addUserId()` that can be called several times? That makes it easier to send one notification to multiple persons, but not to all.

File: README.md [hunk 3]
```diff
@@ -792,21 +782,50 @@ See [openhab-js : actions.Voice](https://openhab.github.io/openhab-js/actions.ht
...
+  - `.withOnClickAction(onClickAction)`
+  - `.withMediaAttachmentURL(mediaAttachmentURL)`: Sets the URL of a media attachment to be displayed with the notification. This URL must be reachable by the push notification client.
+  - `.withActionButton1(actionButton1)`
+  - `.withActionButton2(actionButton2)`
+  - `.withActionButton3(actionButton3)`
```
Comment (mueller-ma) [inline README.md:808]: I'd prefer `addActionButton()`, so the same function can be called up to three times. Maybe raise an exception when called more than three times?

File: README.md (hunk 1 above)
Comment (florian-h05) [reply README.md:800]: Good idea!

File: README.md (hunk 2 above)
Comment (florian-h05) [reply README.md:803]: Sounds good.

File: README.md (hunk 3 above)
Comment (florian-h05) [reply README.md:808]: Sounds good.

File: README.md (hunk 1 above)
Comment (florian-h05) [reply README.md:800]: But please note that media attachment cannot be set for log notifications with the current NotificationActions in https://github.com/openhab/openhab-addons/issues/16934.

**Comment by florian-h05** [comment]
@mueller-ma I have implemented your proposals, updated the unit tests, updates the README and added the missing JSDoc.

**Comment by florian-h05** [comment]
@digitaldan Do you want to review or should I self-merge?

File: README.md (hunk 1 above)
Comment (mueller-ma) [reply README.md]: @digitaldan Maybe drop the three user-facing types (log, broadcast, normal) completely and make everything a "normal" notification? Then log notifications are just like normal ones with the same properties, but don't push.

**Comment by digitaldan** [comment]
LGTM, thanks !!!!!!

**Review by digitaldan — APPROVED** [review]

**Comment by florian-h05** [comment]
I will test this soon in production and then publish a new release and update the add-on.

**Comment by florian-h05** [comment]
Works fine for me, however I can only test the message due to client limitations (iOS App).
Would be great if you could give it a try once I published a new version.

**Comment by digitaldan** [comment]
Will try it today!   I'm hoping to get a test flight build soon of the new IOS app.

**Pull request merged**